### ISSUE TITLE
`x build` lives blissfully assuming llvm does not exist

### ISSUE THREAD

**Issue by BoxyUwU** [rust-lang/rust#110692]
When running `x b` I get the following output:
`Warning: no codegen-backends config matched the requested path to build a codegen backend. Help: add backend to codegen-backends in config.toml.`
after building stage1 library artifacts.
for reference here is my `config.toml`:
```
# Includes one of the default files in src/bootstrap/defaults
profile = "compiler"
changelog-seen = 2

[rust]
debug = true
deny-warnings = false
incremental = false
```
cc @jyn514

**Comment by jyn514** [comment]
cc https://github.com/rust-lang/rust/issues/109610, https://github.com/rust-lang/rust/pull/109642, @lenko-d. I am not sure the fix in #109642, but if you want to keep that approach, you should count LLVM as one of the possible codegen backends.

**Comment by jyn514** [comment]
so, there are two parts to this issue. one is that codegen-backends acts as a hard-off switch: https://github.com/rust-lang/rust/issues/109610
whereas i'd prefer it were a default, the same way that `build.docs` works. because it's a hard switch, the warning is necessary to tell people to enable it in config.toml.

the second part is that the warning is wrong for LLVM, which is built by default and not handled by codegen-backends; Steps which call ensure(CodegenBackends) or the default `x build` shouldn't give a warning, only explicitly calling `x build rustc-codegen-cranelift` from the command line. you could use something like the existing `is_explicitly_invoked` function to determine if that's the case; i think i might have deleted it recently, but you can dig it out of the git history.

i'd prefer to change the option from a hard off to a default, but i'd be okay with just fixing the warning too.

**Comment by xry111** [comment]
I no longer see it with 1.81.0.

**Comment by Kobzol** [comment]
Codegen backends should now be built more explicitly than before, I don't we even have this error message in bootstrap anymore.
